### DISCUSSION TITLE
How does this gem work with subdomains?

### DISCUSSION THREAD

**Discussion by reinvanimschoot** [ElMassimo/js_from_routes#43] [Q&A]
I am trying to convert something like `homeUrl({ subdomain: 'business' })` to this gem. Is there any support for the usage of subdomains?

Cheers!

**Comment by ElMassimo** [comment]
Hi Rein!

These libraries typically deal with paths without the domain, so you'd need to bake something yourself to support adding the origin and subdomain.

You could create your own extension to `definePathHelper` in order to replace the `path` function provided by this library with a function that supports `subdomain`.

```ts
/**
 * Defines a path helper that can make a request or interpolate a URL path.
 *
 * @param {Method} method  An HTTP method
 * @param {string} pathTemplate The path with params placeholders (if any).
 */
export function definePathHelper (method: Method, pathTemplate: string): PathHelper {
  const helper = <T = any>(options?: Options) =>
    request(method, pathTemplate, options) as Promise<T>
  helper.path = (options?: UrlOptions) => formatUrlWithSubdomain(pathTemplate, options)
  helper.pathTemplate = pathTemplate
  return helper
}
```

Then, in Ruby configure [`client_library`](https://js-from-routes.netlify.app/config/#client-library) to target the file where you've defined this helper.
